react-components/src/components/RevealCanvas/ViewerContext.tsx

```diff
@@ -0,0 +1,52 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+import { type Cognite3DViewer } from '@cognite/reveal';
+import { createContext, type ReactElement, type ReactNode, useContext } from 'react';
+import { type RevealRenderTarget } from '../../architecture/base/renderTarget/RevealRenderTarget';
+import { type CameraStateParameters, useCameraStateControl } from './hooks/useCameraStateControl';
+
+const ViewerContext = createContext<RevealRenderTarget | null>(null);
+
+export const ViewerContextProvider = ({
+  cameraState,
+  setCameraState,
+  value,
+  children
+}: {
+  cameraState?: CameraStateParameters;
+  setCameraState?: (cameraState?: CameraStateParameters) => void;
+  value: RevealRenderTarget | null;
+  children: ReactNode;
+}): ReactElement => {
+  return (
+    <ViewerContext.Provider value={value}>
+      <ViewerControls cameraState={cameraState} setCameraState={setCameraState} />
+      {children}
+    </ViewerContext.Provider>
+  );
+};
+
+const ViewerControls = ({
+  cameraState,
+  setCameraState
+}: {
+  cameraState?: CameraStateParameters;
+  setCameraState?: (cameraState?: CameraStateParameters) => void;
+}): ReactNode => {
+  useCameraStateControl(cameraState, setCameraState);
+  return null;
+};
+
+export const useReveal = (): Cognite3DViewer => {
```

Comment:
This should be in `hooks` folder?

Reply:
I don't think so, the convention we have been using so far, is that hooks that are sort-of exposing values from a Context, are put together with that Context. That's how this hook was placed before this PR as well, in ViewerContext.ts